### ISSUE TITLE
can't install

### ISSUE THREAD

**Issue by aolko** [qodo/TableOfContentsX#1]
MODX Revo 2.6.0
```
Error 22001 executing statement: INSERT INTO `modx_site_snippets` (`source`, `property_preprocess`, `name`, `description`, `editor_type`, `category`, `cache_type`, `snippet`, `locked`, `properties`, `moduleguid`, `static`, `static_file`) VALUES (0, 0, 'TableOfContentsX', 'This snippet takes HTML content and generates a table of contents based on the HTML headers
etc. This can return 2 options:\ntoc - is a
containing a list of the headings that are clickable\ncontent - inputted content with headers with anchors', 0, 12, 0, '/**\n * TableOfContentsX\n *\n * This snippet takes HTML content and generates a table of contents based\n * on the HTML headers
,
etc. Each header should have a HTML anchor\n * if it should be clickable.\n * \n * Based on code by Joost de Valk, submitted on 02/08/2011\n * http://www.westhost.com/contest/php/function/create-table-of-contents/124\n *\n * TableOfContentsX is free software; you can redistribute it and/or modify it\n * under the terms of the GNU General Public License as published by the Free\n * Software Foundation; either version 2 of the License, or (at your option) any\n * later version.\n *\n * TableOfContentsX is distributed in the hope that it will be useful, but WITHOUT ANY\n * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR\n * A PARTICULAR PURPOSE. See the GNU General Public License for more details.\n *\n * @author Stewart Orr @ Qodo Ltd \n * @version 1.0.1\n */\n\n// Parameters/options\n// What should the snippet output? Either \'toc\' or \'content\'\n$options = isset($options) && ($options == \'content\' || $options == \'toc\') ? $options : \'toc\';\n\n// If you would prefer the content to be placeholders - DISABLED for now\n//$toPlaceholder = isset($toPlaceholder) ? $toPlaceholder : FALSE ;\n//$toPlaceholderPrefix = isset($toPlaceholderPrefix) ? $toPlaceholderPrefix . \".\" : \'\' ; // If you want to prefix the placeholders\n\npreg_match_all(\'/([^<]+)<\\/h[1-6]>/i\', $input, $matches, PREG_SET_ORDER);\n\n$anchors = array();\n$toc = \'
\'.\"\\n\";\n$i = 0;\n\n// Content should be MODX input\n$content = $input;\n\nforeach ($matches as $heading) {\n\n   if ($i == 0)\n  $startlvl = $heading[1];\n  $lvl = $heading[1];\n\n $ret = preg_match( \'/id=[\\\'|\"](.*)?[\\\'|\"]/i\', stripslashes($heading[2]), $anchor );\n if ( $ret && $anchor[1] != \'\' ) {\n   $anchor = stripslashes( $anchor[1] );\n $add_id = false;\n  } else {\n $anchor = preg_replace( \'/\\s+/\', \'-\', preg_replace(\'/[^a-z\\s]/\', \'\', strtolower( $heading[3] ) ) );\n  $add_id = true;\n   }\n\n   if ( !in_array( $anchor, $anchors ) ) {\n   $anchors[] = $anchor;\n } else {\n  $orig_anchor = $anchor;\n   $i = 2;\n   while ( in_array( $anchor, $anchors ) ) {\n $anchor = $orig_anchor.\'-\'.$i;\n  $i++;\n }\n $anchors[] = $anchor;\n }\n\n   if ($add_id) {\n $content = substr_replace( $content, \'\'.$heading[3].\'\', strpos( $content, $heading[0] ), strlen( $heading[0] ) );\n    }\n\n   $ret = preg_match( \'/title=[\\\'|\"](.*)?[\\\'|\"]/i\', stripslashes( $heading[2] ), $title );\n if ( $ret && $title[1] != \'\' )\n  $title = stripslashes( $title[1] );\n   else    \n  $title = $heading[3];\n $title = trim( strip_tags( $title ) );\n\n  if ($i > 0) {\n  if ($prevlvl < $lvl) {\n $toc .= \"\\n\".\"
\".\"\\n\";\n    } else if ($prevlvl > $lvl) {\n  $toc .= \'\'.\"\\n\";\n   while ($prevlvl > $lvl) {\n  $toc .= \"
\".\"\\n\".\'\'.\"\\n\";\n $prevlvl--;\n   }\n } else {\n  $toc .= \'\'.\"\\n\";\n   }\n }\n\n   $j = 0;\n   $toc .= \'
\'.$title.\'\';\n   $prevlvl = $lvl;\n\n    $i++;\n}\n\nunset( $anchors );\n\nwhile ( $lvl > $startlvl ) {\n $toc .= \"\\n
\";\n  $lvl--;\n}\n\n$toc .= \'\'.\"\\n\";\n$toc .= \'
\'.\"\\n\";\n\n// Finally output the content\nif ($options == \'content\') {\n    echo $content;\n} else {\n  echo $toc;\n}', 0, 'a:0:{}', '', 0, '') Array ( [0] => 22001 [1] => 1406 [2] => Data too long for column 'description' at row 1 ) 
PHP notice: Array to string conversion
snippet_err_createArray
Error saving vehicle object of class modSnippet; criteria: Array ( [name] => TableOfContentsX ) 
Could not install related objects with locally owned keys for vehicle object of class modCategory; criteria: Array ( [category] => TableOfContentsX ) 
```

**Comment by aolko** [comment]
no progress so far?

**Comment by qodo** [comment]
I have found the cause of this and fixed in version 1.1. Let me know if you have any issues. #5144624
